### ISSUE TITLE
APPLICATION_ROOT setting does not work

### ISSUE THREAD

**Issue by malorambo** [spiral-project/ihatemoney#1123]
Hello !

Trying to setup the app on a Debian 11 machine, I cannot find out how to get the APPLICATION_ROOT setting working with nginx. I set it up with `APPLICATION_ROOT = "accounting"`, but nginx will redirect me to my.domain.tld/accounting/accounting/, and will return a 404 NOT FOUND for every file in `static` (looking for them into `accounting/static/`). I tried to tweak the nginx `conf.d/ihatemoney.conf` file, or even copy the whole `static/` folder to `accounting/static/` but it won't solve anything.

Here is the `/etc/nginx/conf.d/ihatemoney.conf` for information : (almost identical to the default generated file)
```
server {
    server_name my.domain.tld;  # (set up correctly)
    keepalive_timeout 5;

    location /static/ {
        alias   /home/my-user/ihatemoney/lib/python3.9/site-packages/ihatemoney/static/;
    }
    location / {
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header Host $http_host;
        proxy_redirect off;
        proxy_connect_timeout 90;
        proxy_send_timeout 180;
        proxy_read_timeout 180;
        proxy_buffer_size 16k;
        proxy_buffers 8 16k;
        proxy_busy_buffers_size 32k;
        proxy_intercept_errors on;
        if (!-f $request_filename) {
            proxy_pass http://ihatemoney_backend;
            break;
        }
    }
}

upstream ihatemoney_backend {
        server unix:/tmp/ihatemoney.gunicorn.sock;
}
```
Is there any chance I messed up somewhere with nginx ?

**Comment by Glandos** [comment]
You can try with :
```
proxy_pass http://ihatemoney_backend/;
```
With a trailing slash. But I see no `accounting` in you nginx.conf, is this set in the gunicorn configuration?

**Comment by haivala** [comment]
I have same problem but I'm using traefik

**Comment by haivala** [comment]
as I explained already. APPLICATION_ROOT needs to be added in https://github.com/spiral-project/ihatemoney/blob/master/conf/entrypoint.sh

**Comment by zorun** [comment]
Right, that makes sense, fixed in #1189 

@malorambo were you also using Docker / docker-compose?

**Comment by zorun** [comment]
Closing, feel free to reopen if there is still a problem.
